### PR TITLE
feat: deprecate UUID users

### PR DESCRIPTION
**Pull request by luthfifahlevi**
Currently we have 3 identifiers for compass users, that are: UUID, ID (with UUID type too), and email.  The UUID identifier has been used for only validating users for every access Compass API,  while the relationship between tables is using ID, even for _updated_by_ column. For validating users flow, it will try to get a user by UUID first. If it is not found, then it will using _UpsertByEmail_ func that run query:
```
INSERT INTO users (uuid, email, provider) VALUES ($1, $2, $3) ON CONFLICT (email)
DO UPDATE SET uuid = $1, email = $2 WHERE users.uuid IS NULL
RETURNING id
```
As we can see, the “actual” identifier is an email with ON CONFLICT query but WHERE condition using UUID instead.

Therefore, to avoid unnecessary maintaining more column, we can focus validating on email and relationship key on ID, so only keep these two identifier. We don't need UUID column anymore.

Reference why it has UUID before: https://github.com/raystack/compass/issues/90

### CODE REVIEW

File: core/user/user.go [hunk 1]
```diff
@@ -33,6 +32,6 @@ func (u *User) Validate() error {
 type Repository interface {
 	Create(ctx context.Context, u *User) (string, error)
 	GetByEmail(ctx context.Context, email string) (User, error)
-	GetByUUID(ctx context.Context, uuid string) (User, error)
-	UpsertByEmail(ctx context.Context, u *User) (string, error)
+	InsertByEmail(ctx context.Context, u *User) (string, error)
+	GetOrInsertByEmail(ctx context.Context, u *User) (string, error)
```
Comment (haveiss) [inline core/user/user.go:36]: what's the differences with `UpsertByEmail` ?

File: internal/store/postgres/migrations/000020_add_constraint_email_unique.up.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+DO $$
```
Comment (haveiss) [inline internal/store/postgres/migrations/000020_add_constraint_email_unique.up.sql:1]: we don't drop the uuid column on the migration?

File: test/e2e_test/helper_test.go [hunk 3]
```diff
@@ -129,7 +129,7 @@ func (c *Client) makeRequest(method, url string, payload, data interface{}) (err
 	}
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Accept", "application/json")
-	req.Header.Set(IDENTITY_HEADER_KEY_UUID, "compassendtoendtest@gotocompany.com")
+	req.Header.Set(IDENTITY_HEADER_KEY_UUID, "66c4a009-1ea3-4a91-bd95-79c8d4f5027e")
```
Comment (haveiss) [inline test/e2e_test/helper_test.go:132]: we don't need this anymore right?

File: core/user/user.go (hunk 1 above)
Comment (luthfifahlevi) [reply core/user/user.go:36]: There is no update query in `GetOrInsertByEmail`, only. get the userID if email already exist OR insert a new row if email not found in DB

File: internal/store/postgres/migrations/000020_add_constraint_email_unique.up.sql (hunk 2 above)
Comment (luthfifahlevi) [reply internal/store/postgres/migrations/000020_add_constraint_email_unique.up.sql:1]: Oh forgot to add that, lol although that's our goal haha. Thanks mas for notice it

File: test/e2e_test/helper_test.go (hunk 3 above)
Comment (luthfifahlevi) [reply test/e2e_test/helper_test.go]: yes correct, updated

**Comment by luthfifahlevi** [comment]
> what should we do with existing data that has unit id, but doesn't have email?

what you mean is uuid, right? as our ADR discussion before, we'll migrate it first to email as 3rd step https://docs.google.com/document/d/1FZYnbD51m4ZpAoS-ZxQslAgiPaoMOdlbFu4ontx_Jck/edit?tab=t.0

**Comment by batrov** [comment]
> > what should we do with existing data that has unit id, but doesn't have email?
> 
> what you mean is uuid, right? as our ADR discussion before, we'll migrate it first to email as 3rd step https://docs.google.com/document/d/1FZYnbD51m4ZpAoS-ZxQslAgiPaoMOdlbFu4ontx_Jck/edit?tab=t.0

have we checked how our clients interact with compass? what happens if they did not send the email key in the header?

**Comment by luthfifahlevi** [comment]
> > > what should we do with existing data that has unit id, but doesn't have email?
> > 
> > 
> > what you mean is uuid, right? as our ADR discussion before, we'll migrate it first to email as 3rd step https://docs.google.com/document/d/1FZYnbD51m4ZpAoS-ZxQslAgiPaoMOdlbFu4ontx_Jck/edit?tab=t.0
> 
> have we checked how our clients interact with compass? what happens if they did not send the email key in the header?

I already checked the clients that use **UUID only** in headers, and for ODS, it found only 2 users, that are journey team and dex, so will inform them to use email headers instead.
For GTF, could you help to check it @anjali9791 @Gadigeppa-J ?

File: internal/store/postgres/migrations/000021_drop_uuid_users_column.up.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+ALTER TABLE users DROP COLUMN IF EXISTS uuid;
```
Comment (haveiss) [inline internal/store/postgres/migrations/000021_drop_uuid_users_column.up.sql:1]: let's do this on separate PR

**Review by haveiss — APPROVED** [review]
